diff --git a/python.bzl b/python.bzl
index 38b25e7..73c25e7 100644
--- a/python.bzl
+++ b/python.bzl
@@ -23,13 +23,13 @@ def _nixpkgs_python_toolchain_impl(repository_ctx):
         python3_runtime = repository_ctx.attr.python3_repo + ":runtime"

     repository_ctx.file("BUILD.bazel", executable = False, content = """
-load("@bazel_tools//tools/python:toolchain.bzl", "py_runtime_pair")
+load("@rules_python//python:defs.bzl", "py_runtime_pair")
 py_runtime_pair(
     name = "py_runtime_pair",
     py3_runtime = {python3_runtime},
 )
 toolchain(
     name = "toolchain",
     toolchain = ":py_runtime_pair",
-    toolchain_type = "@bazel_tools//tools/python:toolchain_type",
+    toolchain_type = "@rules_python//python:toolchain_type",
     exec_compatible_with = {exec_constraints},
     target_compatible_with = {target_constraints},
 )
